src/System.Security.Cryptography.Pkcs/src/Internal/Cryptography/Pal/AnyOS/ManagedPal.KeyTrans.cs

```diff
@@ -72,57 +72,62 @@ internal byte[] DecryptCek(X509Certificate2 cert, out Exception exception)
-                        }
+                        return DecryptKey(rsa, encryptionPadding, _asn.EncryptedKey.Span, out exception);
+                    }
+                }
+            }
 
-#if netcoreapp
-                        cek = ArrayPool<byte>.Shared.Rent(rsa.KeySize / 8);
+            internal static byte[] DecryptCekCore(
+                X509Certificate2 cert,
+                RSA privateKey,
+                ReadOnlySpan<byte> encrypedKey,
+                string keyEncryptionAlgorithm,
+                out Exception exception)
+            {
+                RSAEncryptionPadding encryptionPadding;
 
-                        if (!rsa.TryDecrypt(_asn.EncryptedKey.Span, cek, encryptionPadding, out cekLength))
-                        {
-                            Debug.Fail("TryDecrypt wanted more space than the key size");
-                            exception = new CryptographicException();
-                            return null;
-                        }
+                switch (keyEncryptionAlgorithm)
+                {
+                    case Oids.Rsa:
+                        encryptionPadding = RSAEncryptionPadding.Pkcs1;
+                        break;
+                    case Oids.RsaOaep:
+                        encryptionPadding = RSAEncryptionPadding.OaepSHA1;
+                        break;
+                    default:
+                        exception = new CryptographicException(
+                            SR.Cryptography_Cms_UnknownAlgorithm,
+                            keyEncryptionAlgorithm);
 
-                        exception = null;
-                        return new Span<byte>(cek, 0, cekLength).ToArray();
-#else
-                        exception = null;
-                        return rsa.Decrypt(_asn.EncryptedKey.Span.ToArray(), encryptionPadding);
-#endif
-                    }
+                        return null;
                 }
-                catch (CryptographicException e)
+
+                if (privateKey != null)
                 {
-                    exception = e;
-                    return null;
+                    return DecryptKey(privateKey, encryptionPadding, encrypedKey, out exception);
                 }
-                finally
+                else
                 {
-                    if (cek != null)
+                    using (RSA rsa = cert.GetRSAPrivateKey())
                     {
-                        Array.Clear(cek, 0, cekLength);
-                        ArrayPool<byte>.Shared.Return(cek);
+                        return DecryptKey(rsa, encryptionPadding, encrypedKey, out exception);
                     }
                 }
             }
         }
 
         private static KeyTransRecipientInfoAsn MakeKtri(
-            byte[] cek,
-            CmsRecipient recipient,
-            out bool v0Recipient)
+                byte[] cek,
+                CmsRecipient recipient,
+                out bool v0Recipient)
```

Comment:
This added indentation seems odd.

Reply:
not sure what happened - I just did C-k C-d